cf/api/service_keys_test.go

```diff
@@ -142,6 +142,33 @@ var _ = Describe("Service Keys Repo", func() {
 		})
 	})
 
+	Describe("get service key", func() {
+		It("makes the right request", func() {
+			setupTestServer(testapi.NewCloudControllerTestRequest(testnet.TestRequest{
+				Method:   "GET",
+				Path:     "/v2/service_keys?q=service_instance_guid:fake-instance-guid;name:fake-service-key-name",
+				Response: serviceKeyDetailResponse,
+			}))
+
+			serviceKey, err := repo.GetServiceKey("fake-instance-guid", "fake-service-key-name")
+			Expect(testHandler).To(HaveAllRequestsCalled())
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(serviceKey.Fields.Guid).To(Equal("fake-service-key-guid"))
+			Expect(serviceKey.Fields.Url).To(Equal("/v2/service_keys/fake-guid"))
+			Expect(serviceKey.Fields.Name).To(Equal("fake-service-key-name"))
+			Expect(serviceKey.Fields.ServiceInstanceGuid).To(Equal("fake-service-instance-guid"))
+			Expect(serviceKey.Fields.ServiceInstanceUrl).To(Equal("http://fake/service/instance/url"))
+
+			Expect(serviceKey.Credentials).To(HaveKeyWithValue("username", "fake-username"))
+			Expect(serviceKey.Credentials).To(HaveKeyWithValue("password", "fake-password"))
+			Expect(serviceKey.Credentials).To(HaveKeyWithValue("host", "fake-host"))
+			Expect(serviceKey.Credentials).To(HaveKeyWithValue("port", float64(3306)))
+			Expect(serviceKey.Credentials).To(HaveKeyWithValue("database", "fake-db-name"))
+			Expect(serviceKey.Credentials).To(HaveKeyWithValue("uri", "mysql://fake-user:fake-password@fake-host:3306/fake-db-name"))
+		})
+	})
+
```

Comment:
What about the case where the right call is made but the key does not exist? I'd expect this is a valid `get service key` that we should test and verify that errors are displayed, something like `service key named ... is not valid` or similar.

Unless, of course, you are testing this case somewhere else.